testdata/p4tc_samples_outputs/calculator_parser.c

```diff
@@ -140,10 +140,10 @@ static __always_inline int run_parser(struct __sk_buff *skb, struct headers_t *h
                 hdr_start = hdr_start_save;
                 ebpf_packetOffsetInBits = ebpf_packetOffsetInBits_save;
             }
-            u32 select_0;
-            select_0 = (((((u32)(((u16)tmp_0.p << 8) | ((u16)tmp_2.four & 0xff)) << 8) & ((1 << 24) - 1)) | (((u32)tmp_4.ver & 0xff) & ((1 << 24) - 1))) & ((1 << 24) - 1));
+            u8 select_0[3];
+            storePrimitive32((u8 *)&select_0, 24, ((((((u32)(((u16)tmp_0.p << 8) | ((u16)tmp_2.four & 0xff)) << 8) & ((1 << 24) - 1)) | (((u32)tmp_4.ver & 0xff) & ((1 << 24) - 1))) & ((1 << 24) - 1))));
```

Comment:
It seems like the arguments are in the wrong order, the definition of `storePrimitive32` is the following:

```
void storePrimitive32(u8 *a, u32 value, int size)
```

So the value should come before the size
Honestly, maybe it's better to change storePrimitive32's definition to:

```
void storePrimitive32(u8 *a, int size, u32 value)
```

since `storePrimitive64` has them in that order, but it's up to you